### PR TITLE
tests: tfm: Add TF-M regression test and PSA archituctere tests

### PR DESCRIPTION
**Pull request by joerchan**
Add the TF-M regression tests and PSA archituctere tests. These test-cases are copied from the zephyr samples with the same name.

By having a copy of them we can add sdk-nrf specific kconfigs instead of having to patch the upstream sample.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

test-sdk-nrf: sdk-nrf-PR-9149

### CODE REVIEW

**Review by gopiotr — COMMENTED** [review]
Please add entry into CODEOWNERS file with path to test `/tests/tfm/` and person/people responsible for this test.

I understand why you place those samples to `tests` directory (to emphasize that those samples will be used as an regression **test**). But on the other hand I recommend to place them into `samples` folder just to follow the project structure. In `tests` directory we have collected tests written with "Ztest" or "Unity" test framework. Those TF-M samples are standalone applications - just like other samples. So, for me it will be cleaner if those samples will be placed in `samples` directory. But of course this is only my opinion - we can discuss about it ;)

**Comment by SebastianBoe** [comment]
"In tests directory we have collected tests written with "Ztest" or "Unity" test framework."
Is this your opinion, or a documented standard?

**Comment by SebastianBoe** [comment]
"Those TF-M samples are standalone applications - just like other samples."

Note that this "sample", is not just like other samples, have a look at the main.c file for instance.

It's quite special.

I am okay with using the samples directory, but I'm curious if there is a documented standard for where things should go.

**Comment by SebastianBoe** [comment]
@gopiotr : Note that the two tests we are adding here do not fit the description here at all:

https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/samples.html#samples

They are not nordic-specific. They are not typical use-cases.

They are regression test suites for system and integration testing of our SDK's HALs integration with TF-M.

File: tests/tfm/tfm_psa_test/sample.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    harness: console
+    harness_config:
+      type: multi_line
+      regex:
+        - "\\*\\*\\*\\*\\* PSA Architecture Test Suite - Version .* \\*\\*\\*\\*\\*"
+        - "TOTAL FAILED *: 0"
+sample:
+  name: "TFM PSA Test"
+
+tests:
```
Comment (torsteingrindvik) [inline tests/tfm/tfm_psa_test/sample.yaml:13]: The current "path" is `tests/sample/tfm/...` for the below entries.

Could you change both `sample.yaml` files to be called `testcase.yaml`, and remove the `sample` prefix from the entries?

**Comment by vili-nordic** [comment]
Commit message: archituctere (spelling)

**Review by Vge0rge — APPROVED** [review]
LGTM

**Comment by gopiotr** [comment]
> "In tests directory we have collected tests written with "Ztest" or "Unity" test framework." Is this your opinion, or a documented standard?

@SebastianBoe - This is current state. **Now** in `test` directory we have only tests written with those frameworks. Approach proposed in this PR is first time when somebody tries to put there non-Ztest, non-Unity "test" and that's why I ask about rationality of such proposition. AFAIK it is not documented in any nRF Connect SDK documentation.

In general I agree that samples in this PR are not "Nordic-specific". I agree with @torsteingrindvik, that to be clear, `sample.yaml` should be renamed into `testcase.yaml` and `sample` option should be removed from those yaml files.

**Review by torsteingrindvik — APPROVED** [review]
Internal CI has passed on this PR.

**Comment by SebastianBoe** [comment]
@mia-ko , @FrancescoSer : Hello, should tests be in the NCS docs? If so I need to figure out how to fix the toctree issue, if not I will rename to README.

File: tests/tfm/tfm_psa_test/testcase.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+common:
+  tags: tfm
+  platform_allow: nrf5340dk_nrf5340_cpuapp_ns nrf9160dk_nrf9160_ns
```
Comment (gopiotr) [inline tests/tfm/tfm_psa_test/testcase.yaml:3]: Please add integration platform option to specify which exactly platforms should be used in Jenkins CI and to help us avoid "silent skips" which are possible when we rely only on `platform_allow` option:
```yaml
integration_platforms:
  - nrf5340dk_nrf5340_cpuapp_ns
  - nrf9160dk_nrf9160_ns
```

File: tests/tfm/tfm_regression_test/testcase.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+common:
+  tags: tfm
+  platform_allow: nrf5340dk_nrf5340_cpuapp_ns nrf9160dk_nrf9160_ns
```
Comment (gopiotr) [inline tests/tfm/tfm_regression_test/testcase.yaml:3]: Please add integration platform option to specify which exactly platforms should be used in Jenkins CI and to help us avoid "silent skips" which are possible when we rely only on `platform_allow` option:
```yaml
integration_platforms:
  - nrf5340dk_nrf5340_cpuapp_ns
  - nrf9160dk_nrf9160_n
```

**Review by gopiotr — CHANGES_REQUESTED** [review]
Please add entry into CODEOWNERS file with path to test `/tests/tfm/` and person/people responsible for this test.
https://github.com/nrfconnect/sdk-nrf/blob/main/CODEOWNERS

**Comment by joerchan** [comment]
@gopiotr CODEOWNERS and integration_platform has been updated. Thank you.

**Comment by joerchan** [comment]
> @mia-ko , @FrancescoSer : Hello, should tests be in the NCS docs? If so I need to figure out how to fix the toctree issue, if not I will rename to README.
> 
> /home/sebo/ncs/nrf/tests/crypto/README.rst /home/sebo/ncs/nrf/tests/drivers/lpuart/README /home/sebo/ncs/nrf/tests/lib/edge_impulse/README /home/sebo/ncs/nrf/tests/subsys/bootloader/bl_storage/README /home/sebo/ncs/nrf/tests/bluetooth/tester/README

As agreed with @mia-ko added the tests/tfm/*/READMEs to the table of contents.

**Review by gopiotr — APPROVED** [review]

File: doc/nrf/samples/samples_tfm.rst [hunk 4]
```diff
@@ -0,0 +1,12 @@
+.. _tfm_samples:
+
+Trusted Firmware M samples
+##########################
```
Comment (mia-ko) [inline doc/nrf/samples/samples_tfm.rst:4]:
```suggestion
Trusted Firmware-M (TF-M) samples
#################################
```

**Review by mia-ko — CHANGES_REQUESTED** [review]
There is no README file for secure_services test, should there be?

We probably need to do some changes on the README files moved from Zephyr, I'll try to go through them later today :)

File: doc/nrf/samples/samples_tfm.rst (hunk 4 above)
Comment (joerchan) [reply doc/nrf/samples/samples_tfm.rst:4]: > There is no README file for secure_services test, should there be?

No, there should not be. Only on these two.

**Review by mia-ko — APPROVED** [review]
Looks ok :)
Created a follow-up JIRA task for improving the docs for the two tests that are moved from Zephyr.

**Pull request merged**